app/models/submission.rb

```diff
@@ -38,15 +39,22 @@ class Submission < ApplicationRecord
   store_accessor :application_payload, :json_file, :other_files
 
   def application_reference
-    request_body["applicationRef"]
+    @application_reference ||= request_body["applicationRef"] || request_body["metadata"]["id"]
   end
 
   def document_link_urls
-    request_body.fetch("documentLinks", []).pluck("documentLink")
+    @document_link_urls ||= request_body.fetch("documentLinks", []).pluck("documentLink")
   end
 
   def source
-    # TODO: this should be dynamic based on the source of the submission
-    "Planning Portal"
+    @source ||= request_body.dig("metadata", "source") || "Planning Portal"
+  end
+
+  def planning_portal?
+    source == "Planning Portal"
+  end
+
+  def planx?
+    source == "PlanX"
```

Comment:
I wondered about using `ActiveSupport::StringEnquirer` here but it might be a little too fiddly to be worthwhile.

```ruby
def source
  @source ||= ActiveSupport::StringInquirer.new((request_body.dig("metadata", "source") || "Planning Portal").delete(" ").underscore)
end

delegate :planning_portal?, :plan_x, to: :source
```